python/dbsp/config.py

```diff
@@ -118,11 +128,50 @@ def run(self) -> DBSPPipeline:
         """
 
         # print("yaml:\n" + self.yaml())
+        attached_connectors = []
+        for (key, input) in self.pipeline_config.inputs.to_dict().items():
+            # TODO: Stream gets added by the attached connector
```

Comment:
@ryzhyk with the new connector API the stream gets injected through the attached connector, shall I remove stream from InputEndpointConfig and OutputEndpointConfig. I wasn't really aware of these types for the UI so I just stumbled upon them in the python API.

Reply:
Yes, I think it's ok to remove them.  This is basically a strongly typed representation of the config yaml, which mirrors what `serde` creates when deserializing the yaml on the server side. Which is why these types don't appear directly in the OpenAPI spec.

One nice thing about your new API is that it doesn't require these hidden types any longer. But now we need to model the new entities (connectors and attachments) in the API, and the 1-1 relationship between pipelines and configs. We can discuss offline.